### ISSUE TITLE
[a11y] Screen reader does not work properly for the IAA screens

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#5052]
**Important details:**

* Reproducible also for IAA screens while logging in;

**Affected versions:**

* Mozilla VPN 2.12.0 (2.202211171926);
* Mozilla VPN 2.12.0 (2.202211171941); 

**Tested Platforms:**

* Windows 11;
* MacOS 11.6;

**Prerequisites:**

* Have the Voice Over enabled (MacOS: Apple menu > System Preferences > Accessibility > VoiceOver > Check "Enable VoiceOver") or other Screen reader opened;
* Have Mozilla installed and opened;
* Be in Email screen;

**Steps to reproduce:**

1. Navigate through the Email screen while using the left and right arrow keys, or Caps+O and Caps+U keys on Windows;
1. Observe the behavior;
1. Enter an email address that is not associated with a Fx account and press the “Continue” button;
1. Navigate through the Password screen while using the left and right arrow keys on MacOS, or Caps+O and Caps+U keys on Windows;
1. Observe the behavior;
  1. Enter a password that will not meet the requirements and press the left key or Caps+U keys and observe the behavior (e.g. “12345678”);
  1. Press the right arrow or Caps+O until the eye icon is in focus and observe the behavior;
  1. Press the right arrow or Caps+O until the ToS and PN links are in focus;
1. Enter a valid password and press the “Create account” button;
1. Navigate through the Verification screen while using the left and right arrow keys, or Caps+O and Caps+U keys on Windows;
1. Observe the behavior;

**Expected result:**

* For step 2: The screen reader works properly and narrates the whole content of the Email screen;
* For step 5: The screen reader works properly and narrates the whole content of the Password screen;
* 5.b.: The password is no longer displayed in the encrypted form and the eye icon is no longer cut;
* For step 8: The screen reader works properly and narrates the whole content of the Verification screen;

**Actual result:**

* For step 2: “Enter your email address to continue using Mozilla VPN” text between the Mozilla VPN logo and “Enter email” field is not narrated;
* For step 5: “Finish creating your Firefox account to continue to Mozilla VPN” text between the “Change email” button and “Create password” field is not narrated;
* 5.b.: The password is STILL displayed in the encrypted form and the eye icon is STILL cut;
* For step 8: “Open your email and enter the verification code that was sent” text between the “Enter verification code” and text field is not narrated;

**Notes:**

* See the attached video;

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-3387)
┆Reporter: Bianca Hidecuti

**Comment by data-sync-user** [comment]
➤ Matt Lichtenstein commented:

Santiago Andrigo Bianca Hidecuti Lesley Norton 

This ticket contains multiple bugs, some of which seem to have already been fixed (even though this ticket is unassigned and in the backlog).

It appears Lesley has resolved (via https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5222 ( https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5222|smart-link )): 

{quote}For step 2: The screen reader works properly and narrates the whole content of the Email screen;

For step 5: The screen reader works properly and narrates the whole content of the Password screen;

5.b.: The password is no longer displayed in the encrypted form and the eye icon is no longer cut;

For step 8: The screen reader works properly and narrates the whole content of the Verification screen;{quote}

and I have partially resolved (via https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5309 ( https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5309|smart-link )): 

{quote}5.c.: ToS and PN links gain focus and are narrated by the screen reader. Both links can be opened using the keyboard;{quote}

I say partially here, because the legal disclaimer text (which contains the ToS and PN links) is now accessible by screen readers, but the links are not tabbable/clickable via keyboard navigation yet

Outstanding work is still: 

{quote}5.a.: The password tooltip gains focus and it is narrated by the screen reader;{quote}

and

{quote}5.c.: ToS and PN links gain focus and are narrated by the screen reader. Both [ToS and PN] links can be opened using the keyboard;{quote}

I propose we track these two bugs separately and allow the rest of this ticket to best QA verified before it gets more confusing 😅

The password tooltip issue already has a separate bug filed https://mozilla-hub.atlassian.net/browse/VPN-3499 ( https://mozilla-hub.atlassian.net/browse/VPN-3499|smart-link ) that I propose we keep open.

The ToS and PN keyboard navigation would need a separate ticket filed

In the meantime, I will assign this to myself and will move it to “merged” once the current changes have been reviewed

What do you think?

**Comment by data-sync-user** [comment]
➤ Matt Lichtenstein commented:

Moved to merged. Separate issues are filed for remaining work

https://mozilla-hub.atlassian.net/browse/VPN-3499 ( https://mozilla-hub.atlassian.net/browse/VPN-3499|smart-link ) 

https://mozilla-hub.atlassian.net/browse/VPN-3566 ( https://mozilla-hub.atlassian.net/browse/VPN-3566|smart-link )

Removed these issues from the ticket description above

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Verified this as fixed on Mozilla VPN 2.13.0 (2.202212301153), while using MacOS 12.6.1 and on Windows 11.

  * For step 2: “Enter your email address to continue using Mozilla VPN” text between the Mozilla VPN logo and “Enter email” field is narrated;
  * For step 5: “Finish creating your Firefox account to continue to Mozilla VPN” text between the “Change email” button and “Create password” field is narrated;
    * 5.b.: The password is no longer displayed in the encrypted form and the eye icon is no longer cut;

  * For step 8: “Open your email and enter the verification code that was sent” text between the “Enter verification code” and text field is narrated;

Attaching postfix video.

!vpn-3387.mov|width=2696,height=1858!

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

Marking this “January 2023” sprint even though it was done on “December 2022” since the appropriate Sprint is closed. Note that it is 2.13, however.
